org.mixedrealitytoolkit.uxcore/Experimental/List/VirtualizedScrollRectList.cs

```diff
@@ -236,18 +254,38 @@ public float Scroll
         /// content area, this callback is called, with an integer representing
         /// the index of the item in the list. This callback is invoked after
         /// the GameObject has been positioned, but just before it is made
-        /// visible.
+        /// visible. When the value is set the list of items get reset.
+        /// <see cref="ResetLayout"/>.
         /// </summary>
-        public Action<GameObject, int> OnVisible { get; set; }
+        public Action<GameObject, int> OnVisible
+        {
+            get => onVisible;
+            set
+            {
+                onVisible = value;
+                // If this is changeing, it means how the items are populated is changing, hence reset.
+                ResetLayout();
+            }
+        }
 
         /// <summary>
         /// When a pooled prefab instance is just removed from visibility on
         /// the scroll content area, this callback is called, with an integer
         /// representing the index of the item in the list. VirtualizedScrollRectList
         /// will make this GameObject invisible for you after this callback is
-        /// invoked.
+        /// invoked. When the value is set the list of items get reset
+        /// <see cref="ResetLayout"/>.
         /// </summary>
-        public Action<GameObject, int> OnInvisible { get; set; }
+        public Action<GameObject, int> OnInvisible
+        {
+            get => onInvisible;
+            set
+            {
+                onInvisible = value;
```

Comment:
I think we should check for a value change here.

```cs
    if (onInvisible != value)
    {
        onInvisible = value;
        // If this is changeing, it means how the items are populated is changing, hence reset.
        ResetLayout();
    }
```

Reply:
That's a good catch. Made the updates & added that to the tests as well.